backend/src/database/migrations/1716420585079-ministry-location-migration.ts

```diff
@@ -0,0 +1,107 @@
+import { MigrationInterface, QueryRunner } from "typeorm"
+
+export class MinistryLocationMigration1716420585079 implements MigrationInterface {
+    name='MinistryLocationMigration1716420585079'
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`alter table personnel add column work_location int4`)
+        
+        await queryRunner.query(`alter table personnel add column home_location int4`)
+        
+        await queryRunner.query(`
+            ALTER TABLE "personnel" ADD CONSTRAINT "FK_276679ced4e823b3b6058335091" FOREIGN KEY ("work_location") REFERENCES "location"("id") ON DELETE NO ACTION ON UPDATE NO ACTION
+        `)
+        
+        await queryRunner.query(`
+            ALTER TABLE "personnel" 
+            ADD CONSTRAINT "FK_75543e02cab09a4849d7fa61ae6" FOREIGN KEY ("home_location") REFERENCES "location"("id") ON DELETE NO ACTION ON UPDATE NO action
+        `)
+        
+        await queryRunner.query(`
+            update personnel 
+            set work_location=subquery.id  
+            from (select l.id, ep2.personnel_id from location l join emcr_personnel ep2 on l.location_name = ep2.work_location ) as subquery
+            where personnel.id = subquery.personnel_id
+        `)
+        await queryRunner.query(`
+            update personnel 
+            set home_location=subquery.id  
+            from (select l.id, ep2.personnel_id from location l join emcr_personnel ep2 on l.location_name = ep2.home_location ) as subquery
+            where personnel.id = subquery.personnel_id
+        `)
+        await queryRunner.query(`
+            update personnel 
```

Comment:
I can see why the emcr ones need to be formatted like this, but for bcws, since the location ids use `id` as well instead of location name, these two can be simplified right?